### PR TITLE
appveyor: fix binary-builder-based builds

### PR DESCRIPTION
**Pull request by vtjnash**
Also add a warning for this failure condition, and respect user settings for USE_BINARYBUILDER_* flags.

### CODE REVIEW

File: Make.inc [hunk 1]
```diff
@@ -924,18 +926,20 @@ else
...
 
-
 # BinaryBuilder options.  We default to "on" for all the projects listed in BB_PROJECTS,
 # but only if contrib/normalize_triplet.py works for our requested triplet
 ifeq ($(shell python $(JULIAHOME)/contrib/normalize_triplet.py $(or $(XC_HOST),$(XC_HOST),$(BUILD_MACHINE)) >/dev/null 2>/dev/null; echo $$?),0)
 USE_BINARYBUILDER := 1
 else
+ifneq ($(shell python $(JULIAHOME)/contrib/normalize_triplet.py x86_64-linux-gnu),x86_64-linux-gnu)
+$(warning python normalize_triplet.py appears to be non-functional, so BinaryBuilder disabled)
+endif
```
Comment (staticfloat) [inline Make.inc]: Why isn’t the “echo $$?” sufficient?

Comment (vtjnash) [reply Make.inc]: It is. This tests something entirely different.

Comment (staticfloat) [reply Make.inc]: Can you give me a situation where the script runs successfully but does not recognize x86_64-Linux-gnu ?

Comment (vtjnash) [reply Make.inc]: this is in the failure branch

Comment (staticfloat) [reply Make.inc]: Ah, oh I see.  Got it, I was missing the `else` just above this.  Thanks.

**Pull request merged**